src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -1397,42 +1397,47 @@ void W3DDisplay::Preload_Texture_Assets(Utf8String texture)
 #define DibSize(lpbi) ((lpbi)->biSize + (lpbi)->biSizeImage + (int)(lpbi)->biClrUsed * sizeof(RGBQUAD))
 
 #define DibPaletteSize(lpbi) (DibNumColors(lpbi) * sizeof(RGBQUAD))
+
 void Create_Bmp_File(const char *filename, const char *data, int width, int height)
 {
 #ifdef BUILD_WITH_D3D8
-    BITMAPINFOHEADER *header = (BITMAPINFOHEADER *)LocalAlloc(LMEM_ZEROINIT, sizeof(BITMAPINFOHEADER));
-    header->biSize = sizeof(BITMAPINFOHEADER);
-    header->biWidth = width;
-    header->biHeight = height;
-    header->biPlanes = 1;
-    header->biBitCount = 24;
-    header->biCompression = 0;
-    header->biSizeImage = 24 * header->biHeight * ((header->biWidth + 7) / 8);
-    header->biClrImportant = 0;
-    HANDLE h = CreateFile(filename, GENERIC_READ | GENERIC_WRITE, 0, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
+    // #BUGFIX Simplify code and avoid LocalAlloc. Use the stack.
+    BITMAPINFOHEADER iheader = { 0 };
+    iheader.biSize = sizeof(BITMAPINFOHEADER);
+    iheader.biWidth = width;
+    iheader.biHeight = height;
+    iheader.biPlanes = 1;
+    iheader.biBitCount = 24;
+    iheader.biCompression = 0;
+    iheader.biSizeImage = 24 * iheader.biHeight * ((iheader.biWidth + 7) / 8);
+    iheader.biClrImportant = 0;
+    HANDLE h =
```

Comment:
If we are refactoring you might as well use the normal c++ operations for this and avoid the win32 api functions.

Reply:
And then you could take this out of the `BUILD_WITH_D3D8` ifdef.

Reply:
This is not possible with STL Port in place. I am not going to change logic here.

Reply:
Hmm you're write we don't have iostreams...
But we do have c api file i/o which would avoid this as well and is used throughout the codebase.

Reply:
Yes it could be rewritten maybe, but that is beyond the scope of this change.

Reply:
Could just use RawFileClass